react/week-2/lesson.md

```diff
@@ -193,270 +181,297 @@ We don't need to worry about changing the DOM ourselves! This is what makes Reac
+  }
+
   return (
     <div>
-      Count: {props.count}
-      <button>Click me!</button>
+      Count is: {count}
+      <button id="click-me" onClick={incrementCount}>
+        Click me!
+      </button>
     </div>
-  )
-};
-
-ReactDOM.render(<Counter count={0} />, document.getElementById('root'));
+  );
+}
 ```
 
-Now we need to use one of the class component super powers - state. That means we'll have to convert our `Counter` component to use a class ([interactive example](https://codesandbox.io/s/pjlro5rop7)):
+This component doesn't work! Every time it re-renders, we reset the `count` variable back to 0. We need something to "remember" what the count value was before the re-render. Because of this, we need to use the method that React provides.
+
+This **doesn't** mean that variables are useless. They are still useful for when we need to calculate a value during a render. State is a variable, but not all variables are state.
+
+### React Hooks
+
+React has built-in functionality for initialising and updating state in our components. We will access state via a React *Hook* called `useState`.
+
+Hooks are a new-ish feature in React. You may find older tutorials that don't use Hooks, but don't panic. The concepts we learn here are the same whether or not you use Hooks. We are looking at Hooks first because they are simpler to learn for beginners.
+
+### Importing `useState`
+
+To be able to access the `useState` Hook, we first need to import it from the React package. Let's look at an example ([interactive example](https://codesandbox.io/s/importing-usestate-hook-8jr9f?file=/src/App.js)):
 
 ```js
-class Counter extends Component {
-  render() {
-    return (
-      <div>
-        Count: {this.props.count}
-        <button>Click me!</button>
-      </div>
-    );
-  }
-}
+import React, { useState } from 'react';
+
+console.log(useState)
 ```
 
-Next we'll change the component to use the count from `this.state` instead of `this.props` ([interactive example](https://codesandbox.io/s/42y7xqj700)):
+If we look at the console, `useState` is just a function. It lives inside the React code that you installed when you created the app.
+
+To reference the `useState` function in our component, we need to import it from the React code. The curly braces around `useState` are a bit like writing:
 
 ```js
-class Counter extends Component {
-  render() {
-    return (
-      <div>
-        Count: {this.state.count}
-        <button>Click me!</button>
-      </div>
-    );
-  }
-}
+let useState = React.useState;
 ```
 
-This code has a bug! `this.state` is initialised as an empty object, and so `this.state.count` is undefined. We need to initialise it from props. We can do this in the class constructor ([interactive example](https://codesandbox.io/s/1oyxx4lzz7)):
+In fact we can just write `React.useState` in our component if we want! But to type a bit less code, we import it (using the curly braces) once and then can just use `useState`.
```

Comment:
Shall we show an example here of the curly braces method?

`import React, { useState } from 'react'`

Reply:
I thought about suggesting that we only use `React.useState`/`React.useEffect` but my sense is that they'll need to know it anyway since lots of tutorials/production code use this import method.